os/provisioning.md

```diff
@@ -0,0 +1,127 @@
+
+[clc]: https://coreos.com/os/docs/latest/configuration.html
+[ct]: https://github.com/coreos/container-linux-config-transpiler/releases
+
+### Ignition Config
+
+The Ignition Config is the configuration format consumed directly by Container Linux. Ignition, the utility inside of Container Linux which is responsible for provisioning the machine, actually fetches and executes the Ignition Config.
+
+Ignition Configs are mostly static, distro-agnostic, and meant to be generated by a machine rather than a human. While they can be written directly by users, it is highly discouraged due to the easy that which errors can be introduced. Instead of writing Ignition Configs directly, users are encouraged to use provisioning tools like Matchbox, which transparently translate Container Linux Configs to Ignition Configs, or use the Config Transpiler directly.
+
+![visual overview of the alternate ct workflows](img/transpile-workflow.svg)
+
+As can be seen above, ct will typically only be manually invoked when users are manually provisioning machines. If a provisioning tool like Matchbox is used, ct will transparently be incorporated into the deployment pipeline. In this scenario, the end user only needs to worry about preparing a Container Linux Config - Ignition and the Ignition are merely an implementation detail and knowledge of their existence isn't necessary.
+
+## Config Transpiler
+
+The Container Linux Config Transpiler is the tool that makes it possible for the end user to remain distanced from the details of configuring Container Linux. It's responsible for transforming a Container Linux Config written by an end user into an Ignition Config to be consumed by instances of Container Linux.
+
+To go back to the previous example of configuring etcd, the following config will configure an etcd cluster using the machine's public and private IP addresses:
+
+```container-linux-config:none
+etcd:
+  advertise_client_urls:       http://{PUBLIC_IPV4}:2379
+  initial_advertise_peer_urls: http://{PRIVATE_IPV4}:2380
+  listen_client_urls:          http://0.0.0.0:2379
+  listen_peer_urls:            http://{PRIVATE_IPV4}:2380
+  discovery:                   https://discovery.etcd.io/<token>
+```
+
+It was alluded to earlier that ct requires information about the target environment before it can transform configs which use templating. If this config is passed to ct without any other arguments, ct fails:
+
+```
+$ bin/ct < example.yml
+error: platform must be specified to use templating
+```
+
+This says that the config takes advantage of templating (e.g. `PUBLIC_IPV4`) but ct was invoked without being told what environment to target.
+
+CT can be invoked again and given Amazon EC2 as an example:
+
+```
+$ bin/ct --platform=ec2 < example.yml
```

Comment:
The platform cannot be explicitly stated in the config (that would defeat the purpose of making it platform independent). What gave you impression that this might be a feature that would be supported?